docs/design/assetgeneration.md

```diff
@@ -16,26 +16,29 @@ An asset is the generic representation of work-item for installer that needs to
 
 The asset would usually follow these steps to generate its output:
 
-1. Load components from disk. (If found, mark them from deletion, we consume our inputs.)
+1. Fetch its parent assets.
 
-2. Load its components from State object given by installer.
+2. Generate the assets either by:
+    *  Using the parent assets
+    *  Loading from on-disk assets
+    *  Loading from state file
 
-3. If both the sources exist,
+3. If any of the parent assets is **dirty** (currently we think all on-disk assets are **dirty**), then use the parent assets to generate and return **dirty**.
```

Comment:
The aside about on-disk assets being dirty needs to stand on its own. It is used by the rest of the steps in determining whether the current asset being fetched is dirty. Alternatively (and what I would recommend), replace all of the parts of the steps that mention the current asset being dirty or not dirty with something that says that the current asset is on-disk or not.

Edit: I see that it is mentioned on its own later in the "Dirty detection" section. Maybe that later section is not needed any more or the guts of this section would be better placed in that section.